Move spawn locations for Promy 3rd floor NM's #1250 (alternate location commented out)

hooksta4 opened the pull request:
There are two spawn locations on floor 3 in Promy-Dem.

<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Adds second spawn location for NM Satiator in Promy Dem. Currently, the elevators are wonky at best and will only take to one island, which coincidently isnt the island the NM is on. Solves issue #1250 

## Steps to test these changes

Progress through Dem by defeating Memory Receptacles. Also I spawn/despawned a bunch of times.

hooksta4 committed:
Added second spawn location for NM Saitator #1250

There are two spawn locations on floor 3 in Promy-Dem.

TeoTwawki commented on sql/mob_spawn_points.sql:
```diff
@@ -3760,7 +3760,7 @@ INSERT INTO `mob_spawn_points` VALUES (16851248,'Gorger','Gorger',20,15.474,-0.5
 -- TODO: Implement Apex Mobs
 
-INSERT INTO `mob_spawn_points` VALUES (16851267,'Satiator','Satiator',21,-205.880,-0.500,268.850,80);
+INSERT INTO `mob_spawn_points` VALUES (16851267,'Satiator','Satiator',21,-44.3170,0,-242.987,80);
```
Didn't really matter which one was the "default". on despawn as long as the function `UpdateNMSpawnPoint()` is called it'll select a new one from the other table.

You will need to edit [the mobs script](https://github.com/LandSandBoat/server/blob/base/scripts/zones/Promyvion-Dem/mobs/Satiator.lua) or your extra row in NM Spawn Points will do nothing, as I see the script currently does not call the function. If you want the original spawn at server start to pick a random place, you'll have to do that in the zone script ~~using `setSpawn()` to set its "home" then `setPos()` to move it there (otherwise it will walk back to the 1st position)~~ _scratch that, call  `UpdateNMSpawnPoint()`on init, that's before the 1st spawn_

hooksta4 replied:
So the bigger issue is the MR portals will NOT send players to other island (where the NM is located at). By changing it to the later, the NM spawns on the island where players are transported. Perhaps comment out the other one until the portals are fixed?

TeoTwawki replied:
yes, commented out is fine, but honestly, I would have focused on why the portal isn't random-ing before touching the NM. The problem isn't the NMs problem, and this is working around an issue without really fixing it. So if #1250 is being closed by this I expect a new issue for the teleport problem. A whole island is inaccessible atm and we're kind of removing the incentive to fix it now. I say kind of because I dunno anyone that seriously camps this thing >.>;

hooksta4 replied:
I understand. This is a temporary fix for a larger problem as stated. Either way, the NM does have 2 spawn points and until the portals are fixed, this would at least make the NM available. I suppose the NM is trivial at post 75, but its pretty useful nonetheless.

TeoTwawki replied:
which problem is larger is debatable and honestly not important right now as it's already agreed that it's fine to comment it out - just make a new tracker issue for the portal please. Otherwise it'll be forgotten.

hooksta4 replied:
Sounds good.

hooksta4 replied:
Well hold on, out of curiosity, I went and looked at the other 2 3rd floor NM's. Appears they too, spawn on the inaccessible island. let me update the other two to put on the accessible island and then can create a more specific issue for teleporters.

hooksta4 committed:
Comment out inaccessible island NM.

Will reopen a new portal ticket specifically for portals.

TeoTwawki commented on sql/nm_spawn_points.sql:
```diff
@@ -252,6 +252,8 @@ INSERT INTO `nm_spawn_points` VALUES (16806227,49,-549.802,-8.944,-24.848);
 INSERT INTO `nm_spawn_points` VALUES (16822423,0,4.609,11.472,-101.235); -- Bugbear Strongman (west)
 INSERT INTO `nm_spawn_points` VALUES (16822427,0,60.728,15.487,-118.886); -- Bugbear Strongman (east)
+INSERT INTO `nm_spawn_points` VALUES (16851267,0,-44.3170,0,-242.987); -- Satiator
+-- INSERT INTO `nm_spawn_points` VALUES (16851267,1,-205.880,-0.500,268.850); -- Satiator on inaccessable island. 
```
This wasn't the part that needs to be put in commented out. As it stands now, neither sql row in this table is at all being used at all. Add the function calls, and leave them in commented out, then this PR is ready to pass.

hooksta4 committed:
Moved Promy 3rd Floor Nm's to portal access floors

Updated/verified all 3 promy zones 3rd floor NM's to now spawn on floor portal transports players. Valid spot(s) just removing the chance to spawn on different island until transports are fixed at a later PR. Will need to uncomment out NM spawn points and add UpdateNMSpawnPoint() in each zone.lua

hooksta4 committed:
Update Promy ID's and Zone's to reflect multiple Spawn locations.

TeoTwawki reviewed (approved):
good enough

EpicTaru commented:
https://github.com/LandSandBoat/server/issues/2591 since I didn't see it referenced yet in this PR.
